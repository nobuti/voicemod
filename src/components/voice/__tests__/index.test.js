import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Voice from "../";

describe("Voice", () => {
  const onChange = jest.fn();

  test("renders properly", () => {
    const { container } = render(
      <Voice id={1} name="Wadus" icon="VoicesVoiceIcon01.png" />
    );

    expect(container).toMatchSnapshot();
    expect(screen.queryByTestId("voice")).toBeInTheDocument();
  });

  test("renders properly when active", () => {
    render(
      <Voice id={1} name="Wadus" active={true} icon="VoicesVoiceIcon01.png" />
    );

    expect(screen.queryByTestId("voice")).toHaveClass("is-active");
  });

  test("renders properly when favorite", () => {
    render(
      <Voice id={1} name="Wadus" favorite={true} icon="VoicesVoiceIcon01.png" />
    );

    expect(screen.queryByTestId("voice")).toHaveClass("is-favorite");
  });

  test("reset button is shown properly", () => {
    render(
      <Voice
        id={123}
        name="Wadus"
        icon="VoicesVoiceIcon01.png"
        onChange={onChange}
      />
    );

    fireEvent.click(screen.getByTestId("voice-fav"));
    expect(onChange).toHaveBeenCalledWith({ favorite: true, id: 123 });
  });
});
