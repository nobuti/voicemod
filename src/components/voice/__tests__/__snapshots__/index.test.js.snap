// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voice renders properly 1`] = `
<div>
  <div
    class="Voice"
    data-testid="voice"
    tabindex="0"
  >
    <div
      class="Voice-image"
    >
      <img
        alt="Wadus"
        src="/VoicesVoiceIcon01.png"
      />
      ;
    </div>
    <div
      class="Voice-name"
    >
      Wadus
    </div>
    <button
      class="Voice-favorite"
      data-testid="voice-fav"
    >
      Fav me
    </button>
  </div>
</div>
`;
